core-rust/core-api-server/src/core_api/handlers/lts/state_account_deposit_behaviour.rs

```diff
@@ -0,0 +1,245 @@
+                .collect::<IndexMap<_, _>>()
+        }),
+    )
+}
+
+fn to_api_resource_preference(preference: ResourcePreference) -> models::ResourcePreference {
+    match preference {
+        AccountResourcePreferenceV1::Allowed => models::ResourcePreference::Allowed,
+        AccountResourcePreferenceV1::Disallowed => models::ResourcePreference::Disallowed,
+    }
+}
+
+fn empty_virtual_account_response(
+    context: &MappingContext,
+    header: &LedgerHeader,
+    badge: Option<ResourceOrNonFungible>,
+    requested_resource_addresses: Option<Vec<(String, ResourceAddress)>>,
+) -> Result<Json<models::LtsStateAccountDepositBehaviourResponse>, ResponseError<()>> {
+    response(
+        context,
+        header,
+        &DefaultDepositRule::Accept, // that's how Engine creates new accounts
+        badge.map(|_| false),        // regardless what badge - it is not on the AD list
+        requested_resource_addresses.map(|requested_resource_addresses| {
+            requested_resource_addresses
+                .into_iter()
+                .map(|(requested_resource_address, resource_address)| {
+                    (
+                        requested_resource_address,
+                        empty_virtual_account_resource_specific_bahaviour(resource_address),
+                    )
+                })
+                .collect()
+        }),
+    )
+}
+
+fn empty_virtual_account_resource_specific_bahaviour(
+    resource_address: ResourceAddress,
+) -> models::ResourceSpecificDepositBehaviour {
+    models::ResourceSpecificDepositBehaviour {
+        resource_preference: None,
+        vault_exists: false,
+        is_xrd: resource_address == XRD,
+        deposit_allowed: true,
+    }
+}
+
+fn response(
+    context: &MappingContext,
+    header: &LedgerHeader,
+    default_deposit_rule: &DefaultDepositRule,
+    badge_of_authorized_depositor: Option<bool>,
+    resource_specific_behaviours: Option<
+        IndexMap<String, models::ResourceSpecificDepositBehaviour>,
+    >,
+) -> Result<Json<models::LtsStateAccountDepositBehaviourResponse>, ResponseError<()>> {
+    Ok(models::LtsStateAccountDepositBehaviourResponse {
+        state_version: to_api_state_version(header.state_version)?,
+        ledger_header_summary: Box::new(to_api_ledger_header_summary(context, header)?),
+        default_deposit_rule: match default_deposit_rule {
+            DefaultDepositRule::Accept => models::DefaultDepositRule::Accept,
+            DefaultDepositRule::Reject => models::DefaultDepositRule::Reject,
+            DefaultDepositRule::AllowExisting => models::DefaultDepositRule::AllowExisting,
+        },
+        badge_of_authorized_depositor,
+        resource_specific_behaviours,
+    })
+    .map(Json)
+}
+
+/// Resolves whether the deposit is allowed, based on raw inputs.
+/// See https://docs-babylon.radixdlt.com/main/scrypto/native-blueprints/accounts.html for a
+/// flow chart.
+fn allows_deposit(
```

Comment:
_(note from author)_

## Dear reviewers,

Please pay special attention to this method.

Reply:
Maybe rename to `allows_try_deposit` to be consistent with the blueprint?

Reply:
It's not a big deal. Pick whichever makes more sense within your context. :)